src/manifold/src/impl.cpp

```diff
@@ -677,19 +625,29 @@ void Manifold::Impl::CreateHalfedges(const Vec<glm::ivec3>& triVerts) {
   Vec<int> ids(numHalfedge);
   auto policy = autoPolicy(numTri);
   sequence(policy, ids.begin(), ids.end());
-  for_each_n(policy, zip(countAt(0), triVerts.begin()), numTri,
-             Tri2Halfedges({halfedge_, edge}));
+  for_each_n(policy, countAt(0), numTri,
+             [this, &edge, &triVerts](const int tri) {
+               const glm::ivec3& verts = triVerts[tri];
+               for (const int i : {0, 1, 2}) {
+                 const int j = (i + 1) % 3;
+                 const int e = 3 * tri + i;
+                 halfedge_[e] = {verts[i], verts[j], -1, tri};
+                 // Sort the forward halfedges in front of the backward ones by
+                 // setting the highest-order bit.
+                 edge[e] = glm::uint64_t(verts[i] < verts[j] ? 1 : 0) << 63 |
+                           ((glm::uint64_t)glm::min(verts[i], verts[j])) << 32 |
+                           glm::max(verts[i], verts[j]);
+               }
+             });
   // Stable sort is required here so that halfedges from the same face are
   // paired together (the triangles were created in face order). In some
   // degenerate situations the triangulator can add the same internal edge in
   // two different faces, causing this edge to not be 2-manifold. These are
   // fixed by duplicating verts in SimplifyTopology.
-  stable_sort(policy, zip(edge.begin(), ids.begin()),
-              zip(edge.end(), ids.end()),
-              [](const thrust::tuple<uint64_t, int>& a,
-                 const thrust::tuple<uint64_t, int>& b) {
-                return thrust::get<0>(a) < thrust::get<0>(b);
-              });
+  stable_sort(
+      policy, ids.begin(), ids.end(),
+      [&edge](const int& a, const int& b) { return edge[a] < edge[b]; });
+
```

Comment:
I am curious if we can skip the `edge` tmp buffer completely, and compute the edge information on-the-fly. Maybe we can try a benchmark later, not critical.

Reply:
Yeah, I never liked that either; I couldn't find another way before, but if you have an idea, that would be great.

Reply:
FYI: I tried this, it is slightly slower.